cms/djangoapps/contentstore/views/item.py

```diff
@@ -103,6 +103,20 @@ def _filter_entrance_exam_grader(graders):
     return graders
 
 
+def set_discussion_toggle(value, xblock, user):
+    """
+    Recursively update all children of given xblock in modulestore
+    """
+    for child in xblock.get_children():
+        if child.category in ["chapter", "sequential"]:
+            set_discussion_toggle(value, child, user)
+        else:
+            child.discussion_enabled = value
+            modulestore().update_item(child, user.id)
+
+    modulestore().update_item(xblock, user.id)
```

Comment:
I don't think this is a good idea. It's going around the standard saving mechanism. i.e. call _save_xblock. This block itself will be saved in the regular course of things, just update child blocks.